xds/src/main/java/com/linecorp/armeria/xds/SubscriberStorage.java

```diff
@@ -79,15 +82,20 @@ boolean unregister(XdsType type, String resourceName, ResourceWatcher<AbstractRe
         return false;
     }
 
-    Map<String, XdsStreamSubscriber> subscribers(XdsType type) {
-        return subscriberMap.getOrDefault(type, Collections.emptyMap());
+    <T extends XdsResource> Map<String, XdsStreamSubscriber<T>> subscribers(XdsType type) {
```

Comment:
That's a good question. In fact, this was the most challenging decision I had to make during the implementation. I considered creating four different subclasses of the XdsStreamSubscriber to avoid the unsafe cast, but I felt it might be excessive.

> Although it may seem like the appropriate parser, watcher, and callbacks is called, I don't think the addition of the generic parameter helps with type-safety since adding and fetching subscribers is force-cast anyways.

As you mentioned, the appropriate parser is used here, and by checking it in just one place, we can ensure the safety of this type cast. If this change is implemented, we can be confident that all other occurrences are also safe. Without this modification, we would need to verify multiple places. I hope that clarifies the rationale.